blog/content/post/2024-12-13-ram-data-decay-research.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Research of RAM data remanence times
+abstract: "Dynamic RAM must be periodically refreshed to maintain its content.
+          JEDEC specifies how often this has to be done to ensure proper
+          operation. In this post, we will see for how long the data can be
+          recovered if it isn't."
+cover: /covers/image-file.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2024-12-13
+archives: "2024"
+
+tags:
+  - testing
+categories:
+  - Miscellaneous
+
+---
+
+Dynamic RAM (DRAM) must be periodically refreshed to maintain its content. If
+any bit is inadvertently swapped, we can say that the memory doesn't work as it
+should. JEDEC (Joint Electron Device Engineering Council, a body that provides
+DDR specification, among other things) specifies how often this has to be done
+to ensure proper operation. However, some people are more interested in how long
+the data may be preserved after DRAM is powered down. In this blog post, we will
+describe how this can be measured, and what results we managed to get on a
+couple of tested platforms.
+
+## Some theory
+
+RAM (Random-Access Memory) is a type of memory that can be accessed at random
+locations with little to no difference in time between different addresses. In
+this document, this term will be used to represent operating system's view of
+memory as one(*) consecutive range of addresses.
+
+DRAM (Dynamic RAM) is a type of RAM that holds the data in capacitors, which
+loses its charge over time and must be periodically refreshed. This is in
+contrast to static RAM (SRAM) which doesn't require refreshing to hold its data.
+The capacitor, together with logic required to access its content, is called a
+DRAM cell. Multiple cells form a grid that is addressable by row and column,
+such grid is called a bank. Each bank (in most commonly available DRAM chips) is
+replied 4 or 8 times in parallel, so each row/column pair addresses 4 or 8 bits
```

Comment:
```suggestion
replicated 4 or 8 times in parallel, so each row/column pair addresses 4 or 8 bits
```

Reply:
https://github.com/3mdeb/news-and-ideas/pull/621/commits/445f12165153db5d3d50208baad9c00cd20168fe